tests/test_iqm_sampler.py

```diff
@@ -99,7 +100,10 @@ def run_request():
 
 @pytest.mark.usefixtures('unstub')
 def test_run_sweep_raises_with_non_physical_names(adonis_sampler, circuit_non_physical):
-    with pytest.raises(ValueError, match='Qubit not on device'):
+    when(adonis_sampler._client).get_quantum_architecture().thenReturn(
+        adonis_sampler._device.metadata.to_architecture()
+    )
+    with pytest.raises(CircuitExecutionError, match='Qubit Alice is not allowed as locus for measure'):
```

Comment:
I would add somewhere that this error comes from `iqm-client` and this is more of an integration test rather than `qiskit-iqm` unit test.